pinot-query-planner/src/main/java/org/apache/pinot/query/validate/Validator.java

```diff
@@ -124,4 +129,19 @@ private static String getColumnName(SqlIdentifier identifier) {
     List<String> names = identifier.names;
     return names.get(names.size() - 1);
   }
+
+  protected void addToSelectList(
+      List<SqlNode> list,
+      Set<String> aliases,
+      List<Map.Entry<String, RelDataType>> fieldList,
+      SqlNode exp,
+      SelectScope scope,
+      final boolean includeSystemVars) {
+    if (exp.getKind() == SqlKind.IDENTIFIER) {
+      if ((((SqlIdentifier) exp).names).stream().anyMatch(s -> s.startsWith("$"))) {
```

Comment:
I'm not sure if this would match all the cases we want to ignore @Jackie-Jiang @xiangfu0.

This method is called once for each field in the calcite table row.

Reply:
Does it work if user explicitly select virtual column?

Reply:
I think so, let's add a test for this.